File: high-scores/high-scores.js
export class HighScores {
  constructor(scores) {
    this.scoresGame = scores;
    /* this.sortedScores = this.scoresGame.sort((a, b) => {
      return b - a;
    }); */
  }

  get scores() {
    return this.scoresGame;
  }

  get latest() {
    return this.scoresGame[this.scoresGame.length - 1];
  }

  get personalBest() {
    let sortedScores = this.scoresGame.sort((a, b) => {
      return b - a;
    });
    return sortedScores[0];
  }

  get personalTopThree() {
    let sortedScores = this.scoresGame.sort((a, b) => {
      return b - a;
    });
    return sortedScores.slice(0, 3);
  }
}

File: ruby/high-scores/high_scores.rb
class HighScores
  def initialize(scores)
    @scoresGame = scores
  end

  def scores
    @scoresGame
  end

  def latest
    @scoresGame[-1]
  end

  def personal_best
    @scoresGame.sort[-1]
  end

  def personal_top_three
    sortedScores = @scoresGame.sort{|a, b| b <=> a}
    sortedScores[0..2]
    end

  def latest_is_personal_best?
    @scoresGame.max == @scoresGame[-1]
  end

end


